tests/test_main.py

```diff
@@ -1,29 +1,71 @@
-from fastapi import FastAPI
+import pytest
 from fastapi.testclient import TestClient
-from gethoney.main import app
+from pydantic import ValidationError
+
+from gethoney.crud import Database
+from gethoney.main import app, get_db
 from gethoney.models import Honeypot
 
+test_db = "file::memory:?cache=shared"
+
+
+def override_db():
+    db = Database(test_db)
+    db.create_honeypot(Honeypot(name="test", url="http://12.12.1.1", description="test"))
+    yield db
+
+
+app.dependency_overrides[get_db] = override_db
+
 client = TestClient(app)
 
 
-def test_read_main():
-    response = client.get("/")
-    assert response.status_code == 200
-    assert response.json() == {"Hello": "World"}
+def test_create_honeypot():
+    honeypot = Honeypot(name="test", url="http://12.12.1.1", description="test")
+    data = honeypot.dict()
+    response = client.post("/honeypots/", json=data)
+    assert response.status_code == 201
+    assert response.json() == data
 
 
-#TODO: Fix assert response as list of Honeypots, not just list
-def test_get_honeypots():
+def test_create_honeypot_invalid_url():
+    with pytest.raises(ValidationError):
+        Honeypot(name="test", url="http://12.1", description="test")
+
+
+def test_list_honeypots():
     response = client.get("/honeypots/")
     assert response.status_code == 200
     assert isinstance(response.json(), list)
 
 
+def test_get_nonexistent_honeypot():
+    response = client.get("/honeypots/100")
+    assert response.status_code == 404
+    assert isinstance(response.json(), dict)
 
-def test_create_honeypots():
-    honeypot = Honeypot(name="test",url="http://12.12.1.1", description="test")
-    data = [honeypot.dict()]
 
-    response = client.post("/honeypots/", json=data)
-    assert response.status_code == 201
-    assert response.json() == data
+def test_get_honeypot():
+    response = client.get("/honeypots/1")
+    assert response.status_code == 200
+    assert isinstance(response.json(), list)
```

Comment:
This shouldn't work, since your get endpoint didn't return a list but a single honeypot object as json. So since this test did not fail, you should double check what your get endpoint is returning